blog/content/blog/cncf-dan.md

```diff
@@ -1,74 +1,196 @@
+have meaning, they needed to invest to make sure test coverage was much higher.
+The problem was not just a lack of tests, but a lack of visibility--it required
+heavy, manual data mining just to calculate coverage in the first place.
+
+To find a solution to this problem, Hippie paired with [Rohan
+Fletcher](https://github.com/rohfle). During a discussion with Hippie and Rohan,
+Dan showed them a disk usage graph for OSX and proposed they use a similar graph
+to visualize Conformance coverage of the Kubernetes API.
+
+![disk_graph](/images/blog_image/disk_graph.png "The actual screenshot Dan
+shared to show his vision") The actual screenshot Dan shared to show his vision
+
+## APISnoop
+
+Rohan started on the project to create what would be
+[APISnoop](https://apisnoop.cncf.io/): a visual insight into Kubernetes test
+coverage. This project was well-received by the community when it was introduced
+at Kubecon Europe 2018.
+
+That same year, [Zach Mandeville](https://github.com/zachmandeville) joined the
+team and took over development of APISnoop. At that point there was no real
+automation for generating Snoop's data. Much was still done by manually looking
+for tests in audit event logs. Zach did a lot of the writing, rewriting, and
+architectural changes to APISnoop, along with driving updates of underlying
+Kubernetes logging so that clear coverage information could be distinguished
+from noise.
+
+While APISnoop worked to show gaps in Conformance coverage, a separate effort
+was started to fill those gaps withq tests. Unfortunately, this effort was slow
+going. After about a year, APISnoop showed very little movement on the graph, as
+the test writing efforts yielded very little results. Within character, Dan
+started to look for other ways to get an increase in coverage happening at a
+rate that satisfied his vision.
+
+Since Hippie and his team had been looking at the Kubernetes API and all its
+underlying parts for almost a year, it was a logical fit for them to step up to
+the test-writing efforts, and work to increase the test writing velocity.
+
+## Test-Writing
+
+The writing of tests was a learning experience for everyone in ii, as well as
+the contributors in the Kubernetes community. Initially, the process was to
+quietly work on tests, ensuring they fit all known requirements and then, once
+they seemed ready, to share it with the community through a PR. These PR's
+invariably sparked feedback and needed revisions, and could lead to discussions
+on whether the particular test meaningfully changed the coverage at all. The
+process for writing, rewriting, and collaborating on tests was slower than
+desired.
+
+Hippie and the ii team-- notably Devan Carpenter, [Caleb
+Woodbine](https://github.com/bobymcbobs), and [Stephen
+Heywood](https://github.com/heyste)-- came up with the mock test concept, using
+org-mode in their own flavor of [Emacs (Humacs)](https://www.humacs.org/)
```

Comment:
Link doesn't seem to work at this moment :/ not sure what's up

Reply:
Perhaps in the mean while, use https://github.com/humacs/humacs ?

Reply:
I think the 'working' redirection is from http://humacs.org to https://github.com/humacs/humacs but I'm fine either way.

Reply:
Ah so http://humacs.org is working but not https://humacs.org